### PR TITLE
[LM-10906] Added API Key Security

### PR DESCRIPTION
**Pull request by TyagiLikeMinds**

### CODE REVIEW

File: packages/integration/lib/src/widgets/lists/feed/feed_list.dart [hunk 1]
```diff
@@ -306,6 +306,7 @@ class _LMFeedListState extends State<LMFeedList> {
                     LMFeedFireAnalyticsEvent(
                       eventName: LMFeedAnalyticsKeys.postDeleted,
                       deprecatedEventName: LMFeedAnalyticsKeysDep.postDeleted,
+                      widgetSource: LMFeedWidgetSource.universalFeed,
```
Comment (AnujLM) [inline packages/integration/lib/src/widgets/lists/feed/feed_list.dart:309]: if feed list will be used for other feeds i.e: user_feed or user_created_comment_feed, then is it right to assign widgetSurce: LMFeedWidgetSource.universalFeed ?

File: packages/integration/lib/src/widgets/lists/feed/feed_list.dart [hunk 2]
```diff
@@ -405,6 +406,7 @@ class _LMFeedListState extends State<LMFeedList> {
             LMFeedFireAnalyticsEvent(
               eventName: LMFeedAnalyticsKeys.postLiked,
               deprecatedEventName: LMFeedAnalyticsKeysDep.postLiked,
+              widgetSource: LMFeedWidgetSource.universalFeed,
```
Comment (AnujLM) [inline packages/integration/lib/src/widgets/lists/feed/feed_list.dart:409]: same for here is it right to assign widgetSurce: LMFeedWidgetSource.universalFeed ?

File: packages/integration/lib/src/widgets/lists/feed/feed_list.dart [hunk 3]
```diff
@@ -527,6 +529,7 @@ class _LMFeedListState extends State<LMFeedList> {
                   LMFeedAnalyticsBloc.instance.add(
                       const LMFeedFireAnalyticsEvent(
                           eventName: LMFeedAnalyticsKeys.postCreationStarted,
+                          widgetSource: LMFeedWidgetSource.universalFeed,
```
Comment (AnujLM) [inline packages/integration/lib/src/widgets/lists/feed/feed_list.dart:532]: check this also, is it right to assign widgetSurce: LMFeedWidgetSource.universalFeed ?

File: packages/integration/lib/src/widgets/lists/feed/feed_list.dart [hunk 4]
```diff
@@ -657,6 +660,7 @@ class _LMFeedListState extends State<LMFeedList> {
           eventName: postViewData.isPinned
               ? LMFeedAnalyticsKeys.postPinned
               : LMFeedAnalyticsKeys.postUnpinned,
+          widgetSource: LMFeedWidgetSource.universalFeed,
```
Comment (AnujLM) [inline packages/integration/lib/src/widgets/lists/feed/feed_list.dart:663]: check this also, is it right to assign widgetSurce: LMFeedWidgetSource.universalFeed ?

File: packages/integration/lib/src/widgets/user/user_feed.dart
Comment (AnujLM) [inline packages/integration/lib/src/widgets/user/user_feed.dart:1]: i think this is the file name which we have missed to rename as user_created_post_listview.dart,

**Review by AnujLM — CHANGES_REQUESTED** [review]

File: packages/integration/lib/src/widgets/lists/feed/feed_list.dart (hunk 1 above)
Comment (TyagiLikeMinds) [reply packages/integration/lib/src/widgets/lists/feed/feed_list.dart:309]: That should be customisable too, will change it

File: packages/integration/lib/src/widgets/lists/feed/feed_list.dart (hunk 3 above)
Comment (divyanshgandhilm) [reply packages/integration/lib/src/widgets/lists/feed/feed_list.dart:532]: Have we cleared this with product team - to modify the keys being sent back within the analytic event?

File: packages/integration/lib/src/views/activity/activity_screen.dart [hunk 5]
```diff
@@ -398,6 +398,7 @@ class _LMFeedActivityScreenState extends State<LMFeedActivityScreen> {
                       LMFeedFireAnalyticsEvent(
                         eventName: LMFeedAnalyticsKeys.postDeleted,
                         deprecatedEventName: LMFeedAnalyticsKeysDep.postDeleted,
+                        widgetSource: LMFeedWidgetSource.activityScreen,
```
Comment (divyanshgandhilm) [inline packages/integration/lib/src/views/activity/activity_screen.dart:401]: Let's clear this with product once

File: packages/integration/lib/src/widgets/lists/feed/feed_list.dart (hunk 3 above)
Comment (TyagiLikeMinds) [reply packages/integration/lib/src/widgets/lists/feed/feed_list.dart:532]: We aren't changing anything in the properties map defined by product team. 
Have aligned this change with @ishaan1607

File: packages/integration/lib/src/utils/builder/widget_utility.dart [hunk 6]
```diff
@@ -252,4 +252,14 @@ class LMFeedWidgetUtility {
       Widget topicSelector, List<LMTopicViewData> selectedTopics) {
     return topicSelector;
   }
+
+  Widget composeScreenHeadingTextfieldBuilder(
+      BuildContext context, TextField headingTextField) {
+    return headingTextField;
+  }
+
+  Widget composeScreenContentTextfieldBuilder(
```
[the remark this answers is not in the document]
Comment (TyagiLikeMinds) [reply packages/integration/lib/src/utils/builder/widget_utility.dart:261]: Will test this on master sdk build, throughly

File: packages/integration/lib/src/views/activity/activity_screen.dart (hunk 5 above)
Comment (TyagiLikeMinds) [reply packages/integration/lib/src/views/activity/activity_screen.dart:401]: We haven't modified any key or property map defined by product.

File: packages/integration/lib/src/widgets/lists/feed/feed_list.dart (hunk 1 above)
Comment (TyagiLikeMinds) [reply packages/integration/lib/src/widgets/lists/feed/feed_list.dart:309]: I looked into the widget and it is for fetching universal feed list, just like you created one for `UserSavedPostList`.

File: packages/integration/lib/src/utils/builder/widget_utility.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (divyanshgandhilm) [reply packages/integration/lib/src/utils/builder/widget_utility.dart:261]: Then make sure we have `copyWith()` in LMTaggingAheadTextField

[the remark this answers is not in the document]
Comment (TyagiLikeMinds) [reply packages/integration/lib/src/utils/builder/widget_utility.dart:272]: Added in the latest PR

File: packages/integration/lib/src/widgets/user/user_feed.dart
Comment (TyagiLikeMinds) [reply packages/integration/lib/src/widgets/user/user_feed.dart:1]: Done

File: packages/integration/lib/src/utils/builder/widget_utility.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (TyagiLikeMinds) [reply packages/integration/lib/src/utils/builder/widget_utility.dart:272]: Fixed in https://github.com/LikeMindsCommunity/likeminds-feed-flutter/pull/43

File: packages/integration/lib/src/widgets/user/user_feed.dart
Comment (TyagiLikeMinds) [reply packages/integration/lib/src/widgets/user/user_feed.dart:1]: Fixed in https://github.com/LikeMindsCommunity/likeminds-feed-flutter/pull/43

**Review by divyanshgandhilm — APPROVED** [review]

**Pull request merged**